simulariumio/data_objects/json_data.py

```diff
@@ -0,0 +1,74 @@
+from typing import Dict
+import json
+import numpy as np
+
+from .agent_data import AgentData
+from .frame_data import FrameData
+from .simularium_file_data import SimulariumFileData
+from .trajectory_data import TrajectoryData
+
+
+class JsonData(SimulariumFileData):
+    def __init__(self, file_name: str, file_contents: str):
+        """
+        This object holds JSON encoded simulation trajectory file's
+        data while staying close to the original file format
+
+        Parameters
+        ----------
+        file_name : str
+            Name of the file
+        file_contents : str
+            A string of the data of an open .simularium file
+        """
+        self.file_name = file_name
+        self.data = json.loads(file_contents)
+        self.n_agents = AgentData.from_buffer_data(self.data).n_agents
```

Comment:
Yeah it's a fairly heavy operation. Where would you propose caching it? This is one of the weird situations where in the binary encoding, we save the number of agents in each frame explicitly, but in the json encoding, we don't. And that matters when we're trying to build frames in binary encoding, which need to have n_agents, for a json encoded simularium trajectory

Reply:
I don't want to belabor this if most of our files are not going to be in json format anyway.  I was sort of wondering if the results of AgentData.from_buffer_data would make sense to just live here as a member variable.  Like would it be useful to stash it here for any other operations?   This is probably a lesser use case I guess so it might not make as much sense to overoptimize but you could also write a "cheaper" function to just count up the agents and skip some of the other processing.

Reply:
Yeah I don't think it's really worthwhile to keep track of the AgentData object. I just wrote a more lightweight function to just compute the n_agents